src/index.ts

```diff
@@ -33,14 +35,16 @@ export type SerializationOptions = {
    * More details here: https://buildkite.com/docs/pipelines/dependencies#defining-explicit-dependencies
    */
   explicitDependencies?: boolean;
+  mutator?: MutatorFn;
 };
 
 export type ToJsonSerializationOptions =
   | {
       explicitDependencies: true;
       cache: StepCache;
+      mutator?: MutatorFn;
     }
-  | { explicitDependencies: false };
+  | { explicitDependencies: false; mutator?: MutatorFn };
```

Comment:
I left a comment about this further below checking if we really need the mutator option here, but in any case this can be expressed a bit simpler:

`{ mutator?: MutatorFn } & ({} | {})`

where the part behind the `&` is the discriminated union of the old code.

Reply:
removed this as mentioned above